controllers/candidate.js

```diff
@@ -0,0 +1,115 @@
+const Candidate = require('../Models/candidatemodel');
+const Job = require('../Models/jobsmodel');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const datadictionary = require('../exports');
+
+
+async function candidateProfile(req, res) {
+    const user = await Candidate.findById(req.user.id);
+    return res.render('candidateprofile' , { user: user });
+}
+
+
+async function candidateSignup(req, res) {
+    try {
+        const candidate = new Candidate({
+            username: req.body.username,
+            email: req.body.email,
+            password: await bcrypt.hash(req.body.password, 10),
+            contact: req.body.contact,
+            organization: req.body.organization
+        });
+        await candidate.save();
+        res.status(datadictionary.ok).redirect('/recruiters/login');
+    } catch (err) {
+        if (err.code === datadictionary.mongoerror) {
+            req.flash('error', datadictionary.userexists);
+            res.render('signup', { messages: req.flash() });
+        } else {
+            next(err);
+        }
+    }
+}
+
+async function createToken(id) {
+    const token = jwt.sign({ id }, 'sahfhasfhakshfkdhfa', { expiresIn: 1000 });
+    return token;
+}
+
+async function checkAuthenticated(req,res,next) {
+    const token = req.cookies.jwt;
+    if (!token) {
+        return res.redirect('/recruiters/login');
+    }
+    try {
+        const user1 = jwt.verify(token, 'sahfhasfhakshfkdhfa');
+        req.user = user1;
+        next();
+    } catch (error) {
+        res.redirect('/recruiters/login');
+    }
+}
+
+async function checkNotAuthenticated(req, res, next) {
+    const token = req.cookies.jwt;
+    if (token) {
+        try {
+            const user1 = jwt.verify(token, 'sahfhasfhakshfkdhfa');
+            req.user = user1;
+            return res.redirect('/candidates/profile');
+        } catch (error) {
+            next();
+        }
+    } else {
+        next(); 
+    }
+
+}
+
+async function candidateLogin(req, res, next) {
+    const email = req.body.email;
+    const password = req.body.password;
+    console.log(email,password)
+    try {
+        const user = await Candidate.findOne({ email: email });
+        console.log(user);
+        if (user) {
+            const auth = await bcrypt.compare(password, user.password);
+            if (auth) {
```

Comment:
Can we extract this logic to a separate function so that we increase readability?